### PR TITLE
Add support for `name` claim splitting

### PR DESCRIPTION
**Pull request by DavisRayM**
# Changes

- Update `Pipfile.lock`
- Replace Travis CI with Github actions
- Add support for splitting the `name` claim into `first_name` & `last_name`  when the fields are not found in the mapped user data
- Update `README.md` to include all possible settings

### CODE REVIEW

File: oidc/viewsets.py [hunk 1]
```diff
@@ -149,6 +152,14 @@ def map_claims_to_model_field(self, user_data) -> dict:
                 data[self.map_claim_to_model[k]] = v
             else:
                 data[k] = v
+
+        # Split the name claim into the `first_name` & `last_name`
```
Comment (DavisRayM) [inline oidc/viewsets.py:156]:
```suggestion
        # Split the name claim into `first_name` & `last_name`
```

File: tests/test_viewsets.py [hunk 2]
```diff
@@ -286,4 +296,22 @@ def test_map_claim_to_model(self):
...
+            self.assertEqual(User.objects.count(), user_count)
+
+            # Name is split into first_name and last_name if both any is not
+            # present
+            mock_func.return_value = {
+                "given_name": "Wrong",
+                "name": "Davis Raym",
+                "sub": "davis@m.com",
+            }
+            data = {"id_token": "saasdrrw.fdfdfdswg4gdfs.sadadsods"}
```
Comment (WinnyTroy) [inline tests/test_viewsets.py:308]: Is this id token value just used for testing?
Should this be in the settings file?

File: README.md [hunk 3]
```diff
@@ -30,16 +30,26 @@ INSTALLED_APPS = [
...
+        "family_name": "last_name",
+        "preferred_username": "username",
+    },
+    "SPLIT_NAME_CLAIM": False, # Whether to split the `name` claim into first_name & last_name if present
+    "USER_UNIQUE_FILTER_FIELD": "username",
+    "USE_SSO_COOKIE": True,
+    "SSO_COOKIE_DATA": "email",
     "SSO_COOKIE_MAX_AGE": None,
     "SSO_COOKIE_DOMAIN": "localhost",
+    "JWT_ALGORITHM": "HS256",
```
Comment (WinnyTroy) [inline README.md:48]: This seems to be a duplicate. It was defined in like 34 just above

File: README.md [hunk 4]
```diff
@@ -30,16 +30,26 @@ INSTALLED_APPS = [
...
     "USE_AUTH_BACKEND": False,
     "AUTH_BACKEND": "",  # Defaults to django.contrib.auth.backends.ModelBackend
-    "USE_RAPIDPRO_VIEWSET": False,
+    "REDIRECT_AFTER_AUTH": "http://localhost:3000",
+    "USER_UNIQUE_FILTER_FIELD": None # Defaults to username
```
Comment (WinnyTroy) [inline README.md:52]: Should we be using this `None` value, or should we be using the `username`?
This also seems to be a duplicate key. It's being introduced in line 43 above

**Review by WinnyTroy — CHANGES_REQUESTED** [review]

File: tests/test_viewsets.py (hunk 2 above)
Comment (DavisRayM) [reply tests/test_viewsets.py:308]: This is just a random mocked value; I don't think it should be in the settings

File: README.md (hunk 3 above)
Comment (DavisRayM) [inline README.md:48]:
```suggestion
```

Comment (DavisRayM) [reply README.md]: Changed in the latest commit

File: README.md (hunk 4 above)
Comment (DavisRayM) [reply README.md:51]: `username` is usually the default. Let me remove this duplicate

Comment (DavisRayM) [inline README.md:51]:
```suggestion
```

**Review by WinnyTroy — APPROVED** [review]

**Pull request merged**